modules/services/dunst.nix

```diff
@@ -55,6 +67,48 @@ in
         home.file.".local/share/dbus-1/services/org.knopwob.dunst.service".source =
           "${pkgs.dunst}/share/dbus-1/services/org.knopwob.dunst.service";
 
+        services.dunst.settings.global.icon_folders =
+          let
+            useCustomTheme =
+              cfg.iconTheme.package != hicolorTheme.package
+              || cfg.iconTheme.name != hicolorTheme.name;
+
+            basePaths = [
+              "/run/current-system/sw"
+              "${config.home.homeDirectory}/.nix-profile"
+              cfg.iconTheme.package
+            ] ++ optional useCustomTheme hicolorTheme.package;
+
+            themeNames = [
+              cfg.iconTheme.name
+            ] ++ optional useCustomTheme hicolorTheme.name;
+
+            categoryDirs = [
+              "actions"
+              "animations"
+              "apps"
+              "categories"
+              "devices"
+              "emblems"
+              "emotes"
+              "filesystem"
+              "intl"
+              "mimetypes"
+              "places"
+              "status"
+              "stock"
+            ];
+          in
+            concatStringsSep ":" (
+              concatMap (themeName:
+                concatMap (basePath:
+                  map (category:
+                    "${basePath}/share/icons/${themeName}/32x32/${category}"
```

Comment:
Not sure if it is OK to hard code 32x32 but I'm reluctant to make this option too complicated. This whole icon lookup thing seems to me like an issue that should be resolved inside Dunst itself.

Reply:
Since I have seen `16x16` was used in tutorials, It would be awesome if you could introduce it as option